cpp/modmesh/buffer/pymod/wrap_SimpleArrayPlex.cpp

```diff
@@ -53,6 +53,32 @@ class MODMESH_PYTHON_WRAPPER_VISIBILITY WrapSimpleArrayPlex : public WrapBase<Wr
                     { return wrapped_type(make_shape(shape), datatype); }),
                 pybind11::arg("shape"),
                 pybind11::arg("dtype"))
+            .def_timed(
+                // Note we use double instead of T to avoid the template (,which is not supported by pybind11::init)
+                // Value will later be converted to the correct datatype in the constructor
+                // This may cause the loss of precision, but should be fine to most cases
+                pybind11::init(
+                    [](pybind11::object const & shape, double const & value, std::string const & datatype)
```

Comment:
I decided to use `double` for `value`, please read the comment of explanation.

Reply:
@yungyuc  I am not sure if using `py::object` for `value` is good or not.
here is my concern:
1. if we extract the data type at wrapper, we need to have template `T` for  `SimpleArrayPlex(const shape_type & shape, const T & value, const std::string & data_type)`, then we need to either (1) expose the implementation in the header (2) have many implementation for each data type `T`
2. if we don't extract the `py::object`, and pass it to `SimpleArrayPlex`, `SimpleArrayPlex` can pass the `py::object` to `SimpleArray`, and since `SimlpeArray<T>` already has the data type, then we can extract the `py::object` to the correct data type. However, the drawback is that we need to include pybind11 in `SimpleArray.hpp`

What do you think?

Reply:
The new function `explicit SimpleArrayPlex(const shape_type & shape, double value, const std::string & data_type)` does not seem to be necessary.

You may use [`explicit SimpleArrayPlex(const shape_type & shape, DataType data_type)`](https://github.com/solvcon/modmesh/blob/48ace7cec150185529022673a1e438fba9bb5095/cpp/modmesh/buffer/SimpleArray.hpp#L704) followed by [`SimpleArray::fill(value_type const & value)`](https://github.com/solvcon/modmesh/blob/48ace7cec150185529022673a1e438fba9bb5095/cpp/modmesh/buffer/SimpleArray.hpp#L109) to do the same thing.

Conversion from `py::object` should happen at pybind11 wrapper level because `SimpleArrayPlex` should not know about `PyObject`.  Because `SimpleArray::fill()` does not know `py::object`, it probably saves some coding by providing a multiplexer from `py::object` to `SimpleArray::value_type` at the pybind11 layer.

Reply:
fixed.